diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheckTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheckTest.java
index 8f87eb567..6190c4b20 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheckTest.java
@@ -21,10 +21,12 @@ package com.puppycrawl.tools.checkstyle.checks.modifier;
 
 import static com.google.common.truth.Truth.assertWithMessage;
 import static com.puppycrawl.tools.checkstyle.checks.modifier.RedundantModifierCheck.MSG_KEY;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 import org.junit.jupiter.api.Test;
 
 import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.DetailAstImpl;
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
 import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
 
@@ -126,6 +128,7 @@ public class RedundantModifierCheckTest
 
         final String[] expected = {
             "22:17: " + getCheckMessage(MSG_KEY, "public"),
+            "24:13: " + getCheckMessage(MSG_KEY, "public"),
             "26:21: " + getCheckMessage(MSG_KEY, "public"),
             "37:12: " + getCheckMessage(MSG_KEY, "public"),
             "45:17: " + getCheckMessage(MSG_KEY, "public"),
@@ -149,12 +152,28 @@ public class RedundantModifierCheckTest
             TokenTypes.CLASS_DEF,
             TokenTypes.ENUM_DEF,
             TokenTypes.RESOURCE,
+            TokenTypes.ANNOTATION_DEF,
         };
         assertWithMessage("Invalid acceptable tokens")
             .that(actual)
             .isEqualTo(expected);
     }
 
+    @Test
+    public void testWrongTokenType() {
+        final RedundantModifierCheck obj = new RedundantModifierCheck();
+        final DetailAstImpl ast = new DetailAstImpl();
+        ast.initialize(TokenTypes.LITERAL_NULL, "null");
+
+        final IllegalStateException exception = assertThrows(IllegalStateException.class, () -> {
+            obj.visitToken(ast);
+        }, "IllegalStateException was expected");
+
+        assertWithMessage("Expected and actual violation messages do not match")
+                .that(exception.getMessage())
+                .isEqualTo("Unexpected token type: " + ast.getType());
+    }
+
     @Test
     public void testGetRequiredTokens() {
         final RedundantModifierCheck redundantModifierCheckObj = new RedundantModifierCheck();
@@ -273,6 +292,22 @@ public class RedundantModifierCheckTest
             "13:5: " + getCheckMessage(MSG_KEY, "static"),
             "13:12: " + getCheckMessage(MSG_KEY, "public"),
             "16:9: " + getCheckMessage(MSG_KEY, "public"),
+            "19:5: " + getCheckMessage(MSG_KEY, "public"),
+            "19:12: " + getCheckMessage(MSG_KEY, "static"),
+            "22:5: " + getCheckMessage(MSG_KEY, "public"),
+            "22:12: " + getCheckMessage(MSG_KEY, "abstract"),
+            "22:21: " + getCheckMessage(MSG_KEY, "static"),
+            "26:1: " + getCheckMessage(MSG_KEY, "abstract"),
+            "28:5: " + getCheckMessage(MSG_KEY, "public"),
+            "28:12: " + getCheckMessage(MSG_KEY, "static"),
+            "32:9: " + getCheckMessage(MSG_KEY, "public"),
+            "32:16: " + getCheckMessage(MSG_KEY, "static"),
+            "34:13: " + getCheckMessage(MSG_KEY, "public"),
+            "34:20: " + getCheckMessage(MSG_KEY, "static"),
+            "37:13: " + getCheckMessage(MSG_KEY, "public"),
+            "37:20: " + getCheckMessage(MSG_KEY, "static"),
+            "40:13: " + getCheckMessage(MSG_KEY, "public"),
+            "40:20: " + getCheckMessage(MSG_KEY, "static"),
         };
         verifyWithInlineConfigParser(getPath(
                 "InputRedundantModifierNestedDef.java"), expected);
