diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
index 6bc93617a..530385cc5 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
@@ -57,7 +57,9 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
 
     @Test
     public void testInlineCorrect() throws Exception {
-        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        final String[] expected = {
+            "112: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
+        };
 
         verifyWithInlineConfigParser(
                 getPath("InputSummaryJavadocInlineCorrect.java"), expected);
@@ -81,7 +83,6 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
             "142: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
             "147: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "150: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "155: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
         };
         verifyWithInlineConfigParser(
                 getPath("InputSummaryJavadocIncorrect.java"), expected);
@@ -100,7 +101,7 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
             "80: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
             "94: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
             "108: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "115: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "114: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
         };
         verifyWithInlineConfigParser(
                 getPath("InputSummaryJavadocInlineForbidden.java"), expected);
@@ -148,6 +149,23 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
                 getPath("InputSummaryJavadocIncorrect2.java"), expected);
     }
 
+    @Test
+    public void testIncorrectUsageOfSummaryTag() throws Exception {
+        final String[] expected = {
+            "34: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "41: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "49: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
+            "57: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "64: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
+            "74: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            // Until https://github.com/checkstyle/checkstyle/issues/11425
+            "82: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+        };
+
+        verifyWithInlineConfigParser(
+            getPath("InputSummaryJavadocIncorrect3.java"), expected);
+    }
+
     @Test
     public void testInlineDefaultConfiguration() throws Exception {
         final String[] expected = {
